torch/csrc/distributed/autograd/context/container.cpp

```diff
@@ -160,11 +160,21 @@ void DistAutogradContainer::sendReleaseContextRpc(int64_t context_id) {
   // notify other workers to clean up their contexts.
   auto workerIds =
       autograd_context_.find(context_id)->second->getKnownWorkerIds();
-  auto agent = rpc::RpcAgent::getCurrentRpcAgent();
-  for (const auto& worker_id : workerIds) {
-    agent->send(
-        agent->getWorkerInfo(worker_id),
-        CleanupAutogradContextReq(context_id).toMessage());
+  // agent.send() or getCurrentRpcAgent may throw an error in the case of an
+  // ungraceful shutdown, where we are shutting down RPC and also processing
+  // this message in a separate thread concurrently. In this case, don't throw
+  // here.
+  try {
```

Comment:
Would it be better to move this try/catch inside the for loop? that way, if one of them fails, we'll at least attempt the others.

Reply:
@jjlilley I would like to do that, though the only reason I've seen this failing and what this was supposed to fix is when this is called while RPC is shutdown, in which we have the following 2 errors:
1) The call to `getCurrentAgent()` fails, which is outside the loop
2) `agent.send()` fails since RPC has been shutdown

So even if (2) fails inside the loop, we're certain that the rest will also fail since RPC is shutdown. 

We can probably have separate try/catches for getting the current RPC agent and then try/catches for the individual sends, I can do this in a follow up diff.

Reply:
Sure, go ahead and land this then when the tests clear, so that we can minimize merge conflicts.